website/docs/advanced-concepts/access-control/roles.md

```diff
@@ -0,0 +1,237 @@
+
+
+# Roles
+ A Role is a collection of permissions that defines access for entities or resources which can be further assigned to a user or a group. Roles allow you to adjust multiple permissions rather than assigning individual permissions to users/groups. In Appsmith, a role can contain multiple permissions and a permission can be present in multiple roles. However, a role can't be nested, i.e, it can contain only permissions, not other roles. 
+ 
+ In an Appsmith instance, there are three roles available by default -
+
+ 1. **Admin**
+ 2. **Developer**
+ 3. **App Viewer**
+
+## Resource
+
+When you open a role, the permissions are divided into different tabs based on the resource they affect. Currently, there are four resources -
```

Comment:
Here we should introduce `Resource` as a concept and then give a one-liner definition for each. Something like below:

A resource refer to specific components or aspects of an application that can be accessed or modified by users. These resources may include individual pages or views within the application, specific data sets or database tables, or specific actions or functions that can be performed within the application. In Appsmith, resources are classified into the following categories:

App Resources: These are resources related to the applications built in Appsmith, such as individual pages or views within the application.
Datasource and Queries: These are resources related to databases or queries created in Appsmith, such as a MongoDB setup used as a data source.
Groups and Roles: These are resources related to the roles available for each workspace in Appsmith. For example, if you have two workspaces - HR and Finance - then roles related to HR and finance would be available.
Others: These are resources related to workspaces and audit logs in Appsmith.

Reply:
Thanks @jnikhila , I was yet to work on this. This helps a lot!